crates/ruma-events/src/call/member.rs

```diff
@@ -0,0 +1,386 @@
+//! Types for matrix rtc state events ([MSC3401]).
+//!
+//! This implements a newer/updated version of MSC3401
+//! [MSC3401](https://github.com/matrix-org/matrix-spec-proposals/pull/3401)
+
+use std::time::Duration;
+
+use as_variant::as_variant;
+use ruma_common::{MilliSecondsSinceUnixEpoch as Timestamp, OwnedUserId};
+use ruma_macros::EventContent;
+use serde::{Deserialize, Serialize};
+use tracing::warn;
+
+/// The member state event for a matrixRTC session
+///
+/// This is the object containing all the data related to a matrix users participation in a
+/// matrixRTC session. It consists of memberships/sessions.
+#[derive(Clone, Debug, Serialize, Deserialize, EventContent, PartialEq)]
+#[cfg_attr(not(feature = "unstable-exhaustive-types"), non_exhaustive)]
+#[ruma_event(type = "org.matrix.msc3401.call.member", kind = State, state_key_type = OwnedUserId)]
+pub struct CallMemberEventContent {
+    ///A list of all the memberships that user currently has in this room.
+    ///(can be multiple ones in cases the user participates with multiple devices or there are
+    ///multiple RTC applications (e.g. a call and a spacial experience) running.)
+    memberships: Vec<Membership>,
+}
+
+impl CallMemberEventContent {
+    /// Creates a new `CallMemberEventContent`.
+    pub fn new(memberships: Vec<Membership>) -> Self {
+        Self { memberships }
+    }
+
+    /// All non expired memberships in this member event
+    ///
+    /// # Arguments
+    ///
+    /// * `origin_server_ts` - optionally the `origin_server_ts` can be passed as a fallback in case
+    ///   the Membership does not contain `created_ts`. (`origin_server_ts` will be ignored if
+    ///   `created_ts` is `Some`)
+    pub fn memberships(&self, origin_server_ts: Option<Timestamp>) -> Vec<&Membership> {
```

Comment:
This should probably be renamed to avoid having the same name as the field. Like `active_memberships`.